Sources/AblyChat/Timeserial.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+
+// Timeserial Protocol
+protocol Timeserial: Sendable {
+    var seriesId: String { get }
+    var timestamp: Int { get }
+    var counter: Int { get }
+    var index: Int? { get }
+
+    func toString() -> String
+    func before(_ timeserial: Timeserial) -> Bool
+    func after(_ timeserial: Timeserial) -> Bool
+    func equal(_ timeserial: Timeserial) -> Bool
+}
+
+// DefaultTimeserial Class
+final class DefaultTimeserial: Timeserial {
+    let seriesId: String
+    let timestamp: Int
+    let counter: Int
+    let index: Int?
+
+    private init(seriesId: String, timestamp: Int, counter: Int, index: Int?) {
+        self.seriesId = seriesId
+        self.timestamp = timestamp
+        self.counter = counter
+        self.index = index
+    }
+
+    // Convert to String
+    func toString() -> String {
```

Comment:
This is a function which gets called automatically in JS. No such thing happens in Swift so this function doesn't really serve any purpose. I’d suggest being more selective about which code to copy from JS.

Reply:
ahh i see, removed